app/src/main/java/ch/epfl/sdp/RegisterFormActivity.java

```diff
@@ -0,0 +1,121 @@
+    Button registerButton;
+    TextView loginButton;
+    FirebaseAuth firebaseAuth;
+    FirebaseFirestore fireStore;
+    String userID;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_registerform);
+
+        txtUsername = findViewById(R.id.username);
+        txtEmail = findViewById(R.id.email);
+        txtPassword = findViewById(R.id.password);
+        txtPasswordConf = findViewById(R.id.passwordconf);
+        registerButton = findViewById(R.id.registerbutton);
+
+        firebaseAuth = FirebaseAuth.getInstance();
+        fireStore = FirebaseFirestore.getInstance();
+
+        //check if the User is already log in or not
+        if(firebaseAuth.getCurrentUser() != null) {
+            startActivity(new Intent(getApplicationContext(), MainActivity.class));
+            finish();
+        }
+    }
+
+    private void onUserRegisterFailed() {
+        Toast.makeText(RegisterFormActivity.this, "User could not be created!", Toast.LENGTH_SHORT).show();
+    }
+
+    public void registerBtn_OnClick(View view) {
+        final String username = txtUsername.getText().toString();
+        final String email = txtEmail.getText().toString().trim();
+        String password = txtPassword.getText().toString().trim();
+
+        if (TextUtils.isEmpty(email)) {
+            txtEmail.setError("email is required");
+            return;
+        }
+        if (TextUtils.isEmpty((password))) {
+            txtPassword.setError("password is required");
+            return;
+        }
+        if (password.length() < 8){
+            txtPassword.setError("password length has to be greater than 8 Characters");
+            return;
+        }
+
+        //register user in the firebase
+        firebaseAuth.createUserWithEmailAndPassword(email, password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+            @Override
+            public void onComplete(@NonNull Task<AuthResult> task) {
+                if(!task.isSuccessful()) {
+                    onUserRegisterFailed();
+                    return;
+                }
+
+                Toast.makeText(RegisterFormActivity.this, String.format("User %s with e-mail %s created!", username, email), Toast.LENGTH_SHORT).show();
+                userID = firebaseAuth.getCurrentUser().getUid();
+                DocumentReference documentReference = fireStore.collection("users").document(userID);
+                Map<String, Object> user = new HashMap<>();
+                user.put("ID", userID);
+                user.put("username", username);
+                user.put("email", email);
+                documentReference.set(user).addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        Log.d("TAG", "onSuccess: user Profile is created for" + userID);

```

Comment:
I don't see the point of logging it as opposed to toasting it: if you display failure using toast, then you can display success using toast